src/data/list/big_operators.lean

```diff
@@ -215,6 +215,30 @@ begin
   exact one_le_mul (hl₁ hd (mem_cons_self hd tl)) (ih (λ x h, hl₁ x (mem_cons_of_mem hd h))),
 end
 
+lemma sum_pos [ordered_add_comm_monoid α] :
+  ∀ (l : list α) (hl : ∀ x ∈ l, (0 : α) < x) (hl₂ : l ≠ []), 0 < l.sum
+| [] _ h := (h rfl).elim
+| [b] h _ := by simpa using h
+| (a :: b :: l) hl₁ hl₂ :=
+begin
+  simp only [forall_eq_or_imp, list.mem_cons_iff _ a] at hl₁,
+  rw list.sum_cons,
+  apply add_pos_of_pos_of_nonneg hl₁.1,
+  apply le_of_lt ((b :: l).sum_pos hl₁.2 (l.cons_ne_nil b)),
+end
+
+-- @[to_additive sum_pos]
+lemma one_lt_prod_of_one_lt [ordered_semiring α] :
+  ∀ (l : list α) (hl : ∀ x ∈ l, (1 : α) < x) (hl₂ : l ≠ []), 1 < l.prod
+| [] _ h := (h rfl).elim
+| [b] h _ := by simpa using h
+| (a :: b :: l) hl₁ hl₂ :=
+begin
+  simp only [forall_eq_or_imp, list.mem_cons_iff _ a] at hl₁,
+  rw list.prod_cons,
+  exact one_lt_mul hl₁.1.le (one_lt_prod_of_one_lt _ hl₁.2 $ cons_ne_nil _ _),
+end
+
```

Comment:
```suggestion
@[to_additive sum_pos]
lemma one_lt_prod_of_one_lt [ordered_comm_monoid α] :
  ∀ (l : list α) (hl : ∀ x ∈ l, (1 : α) < x) (hl₂ : l ≠ []), 1 < l.prod
| [] _ h := (h rfl).elim
| [b] h _ := by simpa using h
| (a :: b :: l) hl₁ hl₂ :=
begin
  simp only [forall_eq_or_imp, list.mem_cons_iff _ a] at hl₁,
  rw list.prod_cons,
  apply one_lt_mul_of_lt_of_le' hl₁.1,
  apply le_of_lt ((b :: l).one_lt_prod_of_one_lt hl₁.2 (l.cons_ne_nil b)),
end
```

Reply:
Oh well! I couldn't get it to work with `ordered_comm_monoid`. Thanks!

Reply:
Well I just reverse to_additived your existing proof pretty much!